app/helpers/admin/analytics_helper.rb

```diff
@@ -67,25 +67,84 @@ def total_number_of_searches
       total_searches.html_safe
     end
 
+    def avg_number_searches_per_visit
+      avg_searches_per_visit =
+        "#{AhoyQueries.get_total_number_of_searches
+        .div(AhoyQueries.get_total_number_of_ahoy_visits)}"
+      avg_searches_per_visit.html_safe
+    end
+
     def number_of_visits_initiated_from_homepage
       visits_initiated_from_homepage =
         "#{AhoyQueries.get_number_of_visits_initiated_from_homepage}"
       visits_initiated_from_homepage.html_safe
     end
 
-    def most_visited_locations
-      AhoyQueries.get_most_visited_locations(5)
-    end
-
     def most_used_keywords
       AhoyQueries.get_most_used_keywords(10)
     end
 
-    def avg_number_searches_per_visit
-      avg_searches_per_visit =
-        "#{AhoyQueries.get_total_number_of_searches
-        .div AhoyQueries.get_total_number_of_ahoy_visits}"
-      avg_searches_per_visit.html_safe
+    # calculates the average rating for the given keyword (includes only rated searches)
+    def get_average_search_rating(keyword)
+      all_rated_search_events = AhoyQueries.get_all_rated_search_events_for_keyword(keyword)
+
+      sum_ratings = all_rated_search_events.reduce(0) { |sum, num| sum + num.properties['rating'] }
+
+      number_of_rated_events = all_rated_search_events.length
+      average_rating =
+        if number_of_rated_events.zero?
+          0
+        else
+          sum_ratings/number_of_rated_events
+        end
+
+      [average_rating, number_of_rated_events]
+    end
+
+    def number_of_searches_for_rating(rating)
+      total_searches_for_rating =
+        "(#{AhoyQueries.get_number_of_searches_for_rating(rating)} ratings)"
+
+      total_searches_for_rating.html_safe
+    end
+
+    # retrieves all the search events for the given rating
+    def searches_for_rating(rating)
+      searches_with_rating = AhoyQueries.get_search_events_for_rating(rating)
+
+      group_keywords_and_count(searches_with_rating)
+    end
+
+    # returns a breakdown of the searches with the given keyword rated with the given rating
+    def keyword_search_rating_details(keyword, rating)
+      keyword_searches_with_rating = AhoyQueries.get_keyword_search_rating_details(keyword, rating)
+
+      keyword_search_breakdown = {}
+      keyword_searches_with_rating.each do |search|
+        # get the main category
+        main_category = search.properties['main_category']
+        main_category = '--' if main_category.empty
+
+        # check if any subcategory was applied to the search
+        subcategories = search.properties['subcategories'].empty? ? "No" : "Yes"
+
+        # create a hashmap entry with the main category and 'Yes' or 'No' for subcategories,
```

Comment:
```suggestion
        # create a hashmap entry with the main category and 'Yes' or 'No' to indicate the use of subcategories,
```
What do you think about this?

Reply:
Sounds better than what I had!